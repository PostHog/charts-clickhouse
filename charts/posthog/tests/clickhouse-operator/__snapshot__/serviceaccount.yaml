the manifest should match the snapshot when using default values:
  1: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      name: clickhouse-operator
      namespace: NAMESPACE
