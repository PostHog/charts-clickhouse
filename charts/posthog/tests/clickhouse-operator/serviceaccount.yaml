suite: ClickHouse service account definition
templates:
  - templates/clickhouse-operator/serviceaccount.yaml

tests:
  - it: should be empty if clickhouse.enabled is set to false
    set:
      clickhouse.enabled: false
    asserts:
      - hasDocuments:
          count: 0

  - it: the manifest should match the snapshot when using default values
    asserts:
      - hasDocuments:
          count: 1
      - matchSnapshot: {}

  - it: metadata.namespace override via clickhouse.namespace should work
    set:
      clickhouse.namespace: custom-namespace
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: metadata
          value:
            name: clickhouse-operator
            namespace: custom-namespace
            labels:
              clickhouse.altinity.com/chop: 0.18.4

  - it: metadata.namespace override via Release.Namespace should work
    release:
      namespace: custom-namespace-from-release
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: metadata
          value:
            name: clickhouse-operator
            namespace: custom-namespace-from-release
            labels:
              clickhouse.altinity.com/chop: 0.18.4
