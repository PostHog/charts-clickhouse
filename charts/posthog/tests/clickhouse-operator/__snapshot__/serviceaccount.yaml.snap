the manifest should match the snapshot when using default values:
  1: |
    apiVersion: v1
    kind: ServiceAccount
    metadata:
      labels:
        clickhouse.altinity.com/chop: 0.18.4
      name: clickhouse-operator
      namespace: NAMESPACE
